shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,171 @@
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.DeleteControl;
+import io.javaoperatorsdk.operator.api.reconciler.ErrorStatusHandler;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.RetryInfo;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+@ApplicationScoped
+@ControllerConfiguration(labelSelector = LabelsBuilder.V2_RECONCILER_LABEL_SELECTOR)
+public class ManagedProcessorController implements Reconciler<ManagedProcessor>,
+        EventSourceInitializer<ManagedProcessor>, ErrorStatusHandler<ManagedProcessor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorController.class);
+
+    @ConfigProperty(name = "event-bridge.executor.deployment.timeout-seconds")
+    int executorTimeoutSeconds;
+
+    @ConfigProperty(name = "event-bridge.executor.poll-interval.milliseconds")
+    int executorPollIntervalMilliseconds;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Inject
+    BridgeErrorHelper bridgeErrorHelper;
+
+    @Inject
+    OperatorMetricsService metricsService;
+
+    @Override
+    public List<EventSource> prepareEventSources(EventSourceContext<ManagedProcessor> eventSourceContext) {
+
+        List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(EventSourceFactory.buildResourceInformer(kubernetesClient, LabelsBuilder.V2_OPERATOR_NAME,
+                ManagedProcessor.COMPONENT_NAME,
+                CamelIntegration.class));
+
+        return eventSources;
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        String managedProcessorName = managedProcessor.getMetadata().getName();
+        String managedProcessorNamespace = managedProcessor.getMetadata().getNamespace();
+
+        LOGGER.info("Reconciling ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        ManagedProcessorStatus processorStatus = managedProcessor.getStatus();
+
+        if (!processorStatus.isReady() && isTimedOut(processorStatus)) {
+            // notifyManagerOfFailure
+            processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        String integrationName = String.format("integration-%s", managedProcessorName);
```

Comment:
@lucamolteni Is there any reason not to just use the processor name for the `Integration` name? The `integration-` prefix is kinda irrelevant IMHO as that can be taken from the `kind` of the resource we're creating e.g. `kubectl get Integration $processor_name`.